servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SingleProcessor.java

```diff
@@ -22,28 +22,24 @@
 import io.servicetalk.concurrent.internal.TerminalNotification;
 
 import java.util.Queue;
-import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
 import javax.annotation.Nullable;
 
-import static io.servicetalk.concurrent.internal.ConcurrentUtils.CONCURRENT_EMITTING;
-import static io.servicetalk.concurrent.internal.ConcurrentUtils.CONCURRENT_IDLE;
-import static io.servicetalk.concurrent.internal.ConcurrentUtils.drainSingleConsumerQueueDelayThrow;
-import static io.servicetalk.utils.internal.PlatformDependent.newUnboundedLinkedMpscQueue;
+import static io.servicetalk.utils.internal.PlatformDependent.throwException;
 
 /**
  * A {@link Single} which is also a {@link Subscriber}. State of this {@link Single} can be modified by using the
  * {@link Subscriber} methods which is forwarded to all existing or subsequent {@link Subscriber}s.
  * @param <T> The type of result of the {@link Single}.
  */
 final class SingleProcessor<T> extends Single<T> implements Processor<T, T> {
```

Comment:
Same suggestion as `CompletableProcessor` to remove the "invoke listeners in order" semantics.

Reply:
good point, lets discuss on https://github.com/apple/servicetalk/issues/1033 and handle semantic change in a follow PR.